File: teoricos/tt2/processoAEDSIIVerde/Dona Mônica C/donaMonica.cpp
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

//Prototipação
void leitura(int entrada[][3], int numEntrada);
bool isFim(int entrada[][3], int numEntrada);

int main()
{
    int entrada[1000][3];
    int numEntrada = 0;
    bool fim = false;


    do
    {
        leitura(entrada, numEntrada);
        fim = isFim(entrada, numEntrada);
        numEntrada++;  
    }
    while (fim == false);
    
    numEntrada--;

    int idadeFilhoVelho;

    for(int m = 0; m < numEntrada; m++)
    {
        idadeFilhoVelho = entrada[m][0] - entrada[m][1] - entrada[m][2];
        printf("%i\n", idadeFilhoVelho);
    }


    return 0;
}

void leitura(int entrada[][3], int numEntrada)
{
    bool fim = false;
    int i = 0;
    while(fim == false && i < 3)
    {
        scanf("%i", &entrada[numEntrada][i]);
        i++;
        fim = isFim(entrada, numEntrada);
    }
}

bool isFim(int entrada[][3], int numEntrada)
{
    bool sair;
    if(entrada[numEntrada][0] == 0)
    {
        sair = true;
    }else
    {
        sair = false;
    }

    return sair;
    
}

File: teoricos/tt2/processoAEDSIIVerde/Dona Mônica/donaMonica.java

class donaMonica{

    public static void main (String[] args)
    {
        int[][] entrada = new int[1000][3];
        int numEntrada = 0;
        int i = 0;
        boolean sair = false;
        do
        {
            i = 0;
            while(sair == false && i < 3)
            {
                entrada[numEntrada][i] = MyIO.readInt();
                if(entrada[numEntrada][0] == 0) sair = true;
                i++;
            }
            numEntrada++;

        } while(sair == false);

        numEntrada--; //retira a ultima entrada (fim)

        int idadeFilhoVelho;

        for(int m = 0; m < numEntrada; m++)
        {
            idadeFilhoVelho = 0;
            idadeFilhoVelho = entrada[m][0]- entrada[m][1] - entrada[m][2];
            MyIO.println(idadeFilhoVelho);
        }
    }
}